_layouts/default.html

```diff
@@ -529,29 +529,51 @@ <h3>Why Another Workflow Engine?</h3>
       }
 
       /**
-       * @param {Array} releases - The list of releases.
+       * @param release {{
+       *   name: String,
+       *   published_at: String,
+       *   tag_name: String
+       * }}
+       * @return {string} - Release date formatted as M/D/Y (e.g. 5/14/2020).
+       */
+      function getReleaseDate (release) {
+        return new Date(release.published_at).toLocaleString().split(',')[0]
+      }
+
+      /**
+       * @param {[
+       *   {
+       *     name: String,
+       *     published_at: String,
+       *     tag_name: String
+       *   }
+       * ]} releases - The list of releases.
        */
       function showReleases(releases) {
-        // we only want to iterate over 15 releases...
-        const min = Math.min(releases.length, 15);
-        let foundCylc8Latest = false;
-        let foundCylc7Latest = false;
-        for (let i = 0; i < min; ++i) {
-          const release = releases[i];
-          const name = release.name;
-          const publishedAt = new Date(release.published_at).toLocaleString().split(',')[0];
-          const link = TAGS_URL + release.tag_name;
-          // display latest release metadata
-          if (name.startsWith("cylc-7.9")) {
-            populateLatestReleaseInformation("cylc7", name, link, publishedAt);
-            foundCylc7Latest = true;
-          } else if (name.startsWith("cylc-flow-8")) {
-            populateLatestReleaseInformation("cylc8", name, link, publishedAt);
-            foundCylc8Latest = true;
-          }
-          if (foundCylc7Latest === true && foundCylc8Latest === true) {
-            break;
-          }
+        // show latest Cylc 7 release
+        const latestCylc7 = releases
+          .filter(release => release.name.startsWith("cylc-7.9"))
+          .sort((left, right) => new Date(left.published_at) - new Date(right.published_at))
+          .pop()
+        if (latestCylc7) {
+          populateLatestReleaseInformation(
+            "cylc7",
+            latestCylc7.name,
+            `${TAGS_URL}${latestCylc7.tag_name}`,
+            getReleaseDate(latestCylc7))
+        }
+
+        // show latest Cylc 8 release
+        const latestCylc8 = releases
+          .filter(release => release.name.startsWith("cylc-flow-8"))
+          .sort((left, right) => new Date(left.published_at) - new Date(right.published_at))
+          .pop()
+        if (latestCylc8) {
+          populateLatestReleaseInformation(
+            "cylc8",
+            latestCylc8.name,
+            `${TAGS_URL}${latestCylc8.tag_name}`,
+            getReleaseDate(latestCylc8))
```

Comment:
Iterating arrays is cheap, so we just iterate twice, filtering by the release name, sorting by date asc, and getting the last item with [`.pop()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/pop), which may return undefined.

If not undefined, then we use the old code to populate the HTML elements in the UI.